runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -6937,16 +6937,57 @@ J9::Z::TreeEvaluator::VMifInstanceOfEvaluator(TR::Node * node, TR::CodeGenerator
    return NULL;
    }
 
+TR::Register *
+J9::Z::TreeEvaluator::generateCheckForValueTypeMonitorEnterOrExit(TR::Node *node, TR::Register *objReg, TR::LabelSymbol* mergeLabel, TR::LabelSymbol *helperCallLabel, TR::CodeGenerator *cg)
```

Comment:
We should try to stick to a similar implementation to x86 to avoid complexity and deviation. You only need to send in the `node` and from that you can extract the object register.

Reply:
The reason why I have to pass some extra argument to this function is because in the main evaluator, `callLabel` is either used for inline Recursive implementation of lock routine in OOL or a regular helper call. These label helps me minimizing the OOL section we are generating for monitor node. We certainly do not want to generate 3 OOL for single node, each have their own helper dispatch sequence to call same JIT helper function.

Reply:
Right, I was talking about the need to pass in both `node` and `objReg`, when only `node` will suffice.

Reply:
Fixed in https://github.com/eclipse/openj9/compare/097726868780ab77a37f3f823e2ba1d82900e0db..46e59d80a6c270bd3654d431e7c3372ec3d240b8